test/common/buffer/watermark_buffer_test.cc

```diff
@@ -430,23 +430,34 @@ TEST_F(WatermarkBufferTest, OverflowWatermarkDisabledOnVeryHighValue) {
                                   [&]() -> void { ++overflow_watermark_buffer1; }};
 
   // Make sure the overflow threshold will be above std::numeric_limits<uint32_t>::max()
-  Runtime::LoaderSingleton::getExisting()->mergeValues({{"envoy.buffer.overflow_multiplier", "3"}});
-  buffer1.setWatermarks((std::numeric_limits<uint32_t>::max() / 3) + 1);
-
-  // Add many segments instead of full uint32_t::max to get around std::bad_alloc exception
-  const uint32_t segment_denominator = 128;
-  const uint32_t big_segment_len = std::numeric_limits<uint32_t>::max() / segment_denominator + 1;
-  const std::string big_segment_str = std::string(big_segment_len, 'a');
-  for (uint32_t i = 0; i < segment_denominator; ++i) {
-    buffer1.add(big_segment_str.data(), big_segment_len);
-  }
+  const uint64_t overflow_multiplier = 3;
+  Runtime::LoaderSingleton::getExisting()->mergeValues(
+      {{"envoy.buffer.overflow_multiplier", std::to_string(overflow_multiplier)}});
+  const uint32_t high_watermark_threshold =
+      (std::numeric_limits<uint32_t>::max() / overflow_multiplier) + 1;
+  buffer1.setWatermarks(high_watermark_threshold);
+
+  Buffer::RawSlice iovec;
+  // Reserve and commit uint32_t::max + 1 bytes
+  uint64_t num_reserved =
+      buffer1.reserve(std::numeric_limits<uint32_t>::max() + UINT64_C(1), &iovec, 1);
+  EXPECT_EQ(num_reserved, 1);
+  buffer1.commit(&iovec, 1);
```

Comment:
For the body of the loop:
Buffer::RawSlice iovec[2];
uint64_t num_reserved = buffer1.reserve(big_segment_len, iovec, 2);
  EXPECT_GE(num_reserved, 1);
buffer1.commit(iovec, num_reserved);

Reply:
Done. Also modified the next reservation to use up to 2 iovecs.